plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidDrawAllocationRefactoring.java

```diff
@@ -0,0 +1,170 @@
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.SimpleType;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.core.dom.VariableDeclarationStatement;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+
+import static org.autorefactor.refactoring.ASTHelper.*;
+import static org.eclipse.jdt.core.dom.ASTNode.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.autorefactor.refactoring.ASTBuilder;
+import org.autorefactor.refactoring.Refactorings;
+
+/** See {@link #getDescription()} method. */
+public class AndroidDrawAllocationRefactoring extends AbstractRefactoringRule {
+
+    @Override
+    public String getDescription() {
+        return "Optimization for Android applications to avoid the allocation of"
+                + " objects inside drawing routines. ";
+    }
+
+    @Override
+    public String getName() {
+        return "Android DrawAllocation";
+    }
+
+    @Override
+    public boolean visit(MethodDeclaration node) {
+        if (isMethod(node, "android.widget.TextView", "onDraw", "android.graphics.Canvas")) {
+            final ASTBuilder b = this.ctx.getASTBuilder();
+            final Refactorings r = this.ctx.getRefactorings();
+            node.accept(new OnDrawTransformer(this.ctx, node));
+        }
+        return VISIT_SUBTREE;
+    }
+
+    static class OnDrawTransformer extends ASTVisitor {
+        private RefactoringContext ctx;
+        private MethodDeclaration onDrawDeclaration;
+
+        public OnDrawTransformer(RefactoringContext ctx, MethodDeclaration onDrawDeclaration) {
+            this.ctx = ctx;
+            this.onDrawDeclaration = onDrawDeclaration;
+        }
+
+        public boolean isMethodBindingSubclassOf(ITypeBinding typeBinding, List<String> superClassStrings) {
+            ITypeBinding superClass = typeBinding;
+            while (superClass != null && !superClass.equals(ctx.getAST().resolveWellKnownType("java.lang.Object"))) {
+                String className = superClass.getName();
+                if (className.contains("<")) {
+                    className = className.split("<", 2)[0];
+                }
+                if (superClassStrings.contains(className)) {
+                    return true;
+                }
+                superClass = superClass.getSuperclass();
+            }
+            return false;
+        }
+
+        @Override
+        public boolean visit(VariableDeclarationFragment node) {
+            Expression initializer = node.getInitializer();
+            if (initializer != null) {
+                if (initializer.getNodeType() == ASTNode.CAST_EXPRESSION) {
+                    initializer = ((CastExpression) initializer).getExpression();
+                }
+                if (initializer.getNodeType() == ASTNode.CLASS_INSTANCE_CREATION) {
```

Comment:
Switch will not work as expected. Since ``initializer`` is changed inside the first case, there are cases in which both blocks are executed.